File: frontend/types/payment.ts
export interface CreatePaymentIntentResponse {
  paymentIntentId: string;
  clientSecret: string;
  amount: number;
  currency: string;
  status: string;
}

export interface ConfirmPaymentRequest {
  isSuccess: boolean;
  failureReason?: string;
}

export interface StripeConfigResponse {
  publishableKey: string;
}

export interface Payment {
  id: string;
  appointmentId: string;
  stripePaymentIntentId: string;
  amount: number;
  currency: string;
  status: string;
  failureReason?: string;
  createdAt: string;
  updatedAt: string;
}

File: backend/PaymentService/Models/Payment.cs
namespace PaymentService.Models;

public class Payment
{
    public Guid Id { get; set; }
    public Guid AppointmentId { get; set; }
    public string StripePaymentIntentId { get; set; } = string.Empty;
    public string ClientSecret { get; set; } = string.Empty;
    public long Amount { get; set; }   // cents
    public string Currency { get; set; } = "usd";
    public string Status { get; set; } = "Pending";
    public string? FailureReason { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}